coda-src/kerndep/coda.h

```diff
@@ -149,13 +149,29 @@ struct venus_dirent {
 
 #ifndef _VENUS_DIRENT_T_
 #define _VENUS_DIRENT_T_ 1
+#if defined(__FreeBSD__)
+#include <sys/param.h>
+#endif
+#if defined(__FreeBSD__) && (__FreeBSD_version >= 1200031)
```

Comment:
This dirent layout change is probably a bit harder. I guess this sort of works because of the way the 'world' is built on a FreeBSD system which keeps kernel and user space somewhat better in sync. However the venus_dirent really only by chance used to match the in-kernel layout on FreeBSD. I believe the structure matches whatever NetBSD on Mach used to use.
Linux never had a matching layout in kernel and it reads the data from the venus_dirent and populates the directory data returned to the VFS through a filldir callback. The advantage of that is that the same Coda client binary can be used across all different kernel versions because the translation happens in the kernel.